### PR TITLE
Update templates and workflows to Go 1.20 and deps to latest versions

### PR DESCRIPTION
**Pull request by cnunciato**
- Update all templates and workflows to Go 1.20
- Run generate-go-mod.sh

### CODE REVIEW

**Review by justinvp — APPROVED** [review]
LGTM. Thanks!

**Comment by cnunciato** [comment]
Found a couple of issues these -- the kubernetes-go template for example seems to fail with the upgrade to 4.2.0. Looking into it, tagged do-not-merge for now.

File: serverless-azure-go/app/go.mod [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module fn
 
-go 1.18
+go 1.20
```
Comment (cnunciato) [inline serverless-azure-go/app/go.mod:3]: Need to make sure this also deploys successfully.

**Comment by cnunciato** [comment]
All good -- my branch just hadn't been updated. Will merge once the tests pass.

**Pull request merged**